extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/directives/directive.component.map.js

```diff
@@ -0,0 +1,208 @@
+angular.module('PaperUI.component', []) //
+.directive('mapComponent', function() {
+    var Drag = function(callback) {
+
+        ol.interaction.Pointer.call(this, {
+            handleDownEvent : Drag.prototype.handleDownEvent,
+            handleDragEvent : Drag.prototype.handleDragEvent,
+            handleMoveEvent : Drag.prototype.handleMoveEvent,
+            handleUpEvent : Drag.prototype.handleUpEvent
+        });
+
+        this.coordinate_ = null;
+        this.cursor_ = 'pointer';
+        this.feature_ = null;
+        this.previousCursor_ = undefined;
+        this.callback_ = callback;
+
+    };
+    ol.inherits(Drag, ol.interaction.Pointer);
+
+    Drag.prototype.handleDownEvent = function(evt) {
+        var map = evt.map;
+
+        var feature = map.forEachFeatureAtPixel(evt.pixel, function(feature) {
+            return feature;
+        });
+
+        if (feature) {
+            this.coordinate_ = evt.coordinate;
+            this.feature_ = feature;
+        }
+
+        return !!feature;
+    };
+
+    Drag.prototype.handleDragEvent = function(evt) {
+        var deltaX = evt.coordinate[0] - this.coordinate_[0];
+        var deltaY = evt.coordinate[1] - this.coordinate_[1];
+
+        var geometry = /** @type {ol.geom.SimpleGeometry} */
+        (this.feature_.getGeometry());
+        geometry.translate(deltaX, deltaY);
+
+        this.coordinate_[0] = evt.coordinate[0];
+        this.coordinate_[1] = evt.coordinate[1];
+    };
+
+    Drag.prototype.handleMoveEvent = function(evt) {
+        if (this.cursor_) {
+            var map = evt.map;
+            var feature = map.forEachFeatureAtPixel(evt.pixel, function(feature) {
+                return feature;
+            });
+            var element = evt.map.getTargetElement();
+            if (feature) {
+                if (element.style.cursor != this.cursor_) {
+                    this.previousCursor_ = element.style.cursor;
+                    element.style.cursor = this.cursor_;
+                }
+            } else if (this.previousCursor_ !== undefined) {
+                element.style.cursor = this.previousCursor_;
+                this.previousCursor_ = undefined;
+            }
+        }
+    };
+
+    Drag.prototype.handleUpEvent = function() {
+        this.callback_(this.coordinate_);
+        this.coordinate_ = null;
+        this.feature_ = null;
+        return false;
+    };
+
+    function createMap(element, coordinateCallback) {
+        var mapLayer = new ol.layer.Tile({
+            source : new ol.source.OSM()
```

Comment:
This is indeed a very nice idea. For OH I can see that base layer providers may be exchangeable. For ESH we should stick with OSM. Do you mind creating an issue (feature request) for this?